openmls/src/credentials/mod.rs

```diff
@@ -299,38 +254,35 @@ impl BasicCredential {
     ///
     /// Returns a [`BasicCredentialError`] if the length of the identity is too
     /// large to be encoded as a variable-length vector.
-    pub fn new(identity: Vec<u8>) -> Result<Self, BasicCredentialError> {
-        let mut header = Vec::new();
-        tls_codec::vlen::write_length(&mut header, identity.len())?;
-        Ok(Self { header, identity })
+    pub fn new(identity: Vec<u8>) -> Self {
+        Self {
+            identity: identity.into(),
+        }
     }
 
     /// Get the identity of this basic credential as byte slice.
     pub fn identity(&self) -> &[u8] {
-        &self.identity
+        self.identity.as_slice()
     }
 }
 
 impl From<BasicCredential> for Credential {
-    fn from(mut credential: BasicCredential) -> Self {
-        let mut serialized_credential_content = credential.header;
-        serialized_credential_content.append(&mut credential.identity);
+    fn from(credential: BasicCredential) -> Self {
         Credential {
-            serialized_credential_content: serialized_credential_content.into(),
             credential_type: CredentialType::Basic,
+            serialized_credential_content: credential.identity,
         }
     }
 }
 
-impl TryFrom<&Credential> for BasicCredential {
+impl TryFrom<Credential> for BasicCredential {
     type Error = BasicCredentialError;
 
-    fn try_from(credential: &Credential) -> Result<Self, Self::Error> {
+    fn try_from(credential: Credential) -> Result<Self, Self::Error> {
         match credential.credential_type() {
-            CredentialType::Basic => {
-                let identity: VLBytes = credential.deserialized().unwrap();
-                Ok(BasicCredential::new(identity.into())?)
-            }
+            CredentialType::Basic => Ok(BasicCredential::new(
+                credential.serialized_credential_content.clone().into(),
```

Comment:
This clone should be unnecessary since we own the credential. You can own the `credential_type` in the `match`, that's cheaper.